sources/packages/backend/apps/queue-consumers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -55,4 +60,25 @@ export class StudentAssessmentService {
       workflowName: data.workflowName,
     };
   }
+
+  /**
+   * Retrieve assessment to be retried up to a date.
+   * @param retryMaxDate max date to retrieve assessment to be retried.
+   * @param studentAssessmentStatus student assessment status to be queried.
+   * @returns student assessment to be retried.
+   */
+  async getAssessmentToBeRetried(
+    retryMaxDate: Date,
+    studentAssessmentStatus: StudentAssessmentStatus,
```

Comment:
we can create a type to be more specific 

```ts
type RetryAssessmentStatus =
  | StudentAssessmentStatus.Queued
  | StudentAssessmentStatus.CancellationQueued;
```

Reply:
That was the original suggestion. @andrepestana-aot was there any restriction to have it implemented as suggested?
@dheepak-aot  I am not sure if a type is needed, but I would agree that it can be restricted.

Reply:
My intention was mostly towards restriction

Reply:
Same page @dheepak-aot. Either way would work IMO.